spring-integration-gcp/src/main/java/org/springframework/integration/gcp/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -85,4 +98,12 @@ protected void doStop() {
 
 		super.doStop();
 	}
+
+	public AckMode getAckMode() {
+		return this.ackMode;
+	}
+
+	public void setAckMode(AckMode ackMode) {
+		this.ackMode = ackMode;
```

Comment:
Should be with the `Assert.notNull()` otherwise you have an `NPE` on the `this.ackMode == AckMode.MANUAL` condition above.